marlowe-playground-client/src/SimulationPage/State.purs

```diff
@@ -94,18 +94,39 @@ handleAction (SetIntegerTemplateParam templateType key value) = do
 handleAction StartSimulation =
   void
     $ runMaybeT do
+        -- TODO: move the initialSlot, extended contract and template contract out of the ExecutionState
+        --       and into the SimultationPage top State. This will remove the need of the Maybe
         initialSlot <- MaybeT $ peruse (_currentMarloweState <<< _executionState <<< _SimulationNotStarted <<< _initialSlot)
         extendedContract <- MaybeT $ peruse (_currentMarloweState <<< _executionState <<< _SimulationNotStarted <<< _extendedContract <<< _Just)
         templateContent <- MaybeT $ peruse (_currentMarloweState <<< _executionState <<< _SimulationNotStarted <<< _templateContent)
-        contract <- hoistMaybe $ toCore $ fillTemplate templateContent (extendedContract :: EM.Contract)
-        modifying _marloweState (extendWith (updatePossibleActions <<< updateStateP <<< (set _executionState (emptyExecutionStateWithSlot initialSlot (contract :: S.Contract)))))
+        (contract :: S.Contract) <- hoistMaybe $ toCore $ fillTemplate templateContent (extendedContract :: EM.Contract)
+        -- It is expected that when this function is called, the _marloweState list has a single element
+        -- with an ExecutionState of SimulationNotStarted, and the fact that we are inside a runMaybeT
+        -- with a Prism for _SimulationNotStarted guarantees us that at least the head of the list is
+        -- is in that state. But in theory there could be a list like this
+        -- [SimulationNotStarted, SimulationRunning, SimulationNotStarted] or [SimulationNotStarted, SimulationNotStarted]
+        -- which could result in weird bugs.
+        -- TODO: refactor marloweState type so we can remove "extendWith".
+        --       See note on SimulationPage.Types - Type State {marloweState}
```

Comment:
This method only assumes that the first element of the list is `SimulationNotStarted`, I think the cases you mentioned would work fine, or even if we add an "undo" button to the `SimulationNotStarted` form it would still work as one would expect. But we can indeed prevent them from happening altogether by changing the types